packages/swap/src/utils/getPoolQuote.ts

```diff
@@ -1,25 +1,69 @@
-import { QuoteQueryResponse, SwapFee } from '@/shared/schemas';
-import { buildFee, calculateIncludedSwapFees } from '@/swap/utils/fees';
+import { getAssetAndChain, getInternalAsset } from '@/shared/enums';
+import { getPipAmountFromAmount } from '@/shared/functions';
+import { QuoteQueryResponse } from '@/shared/schemas';
 import { estimateSwapDuration } from '@/swap/utils/swap';
-import logger from './logger';
-import { getMinimumEgressAmount, getEgressFee } from './rpc';
+import { buildFee, getPoolFees } from './fees';
+import { getMinimumEgressAmount } from './rpc';
 import ServiceError from './ServiceError';
-import { getSwapRate } from './statechain';
-import { InternalAsset } from '../client';
+import { LimitOrders, getSwapRateV2 } from './statechain';
+import { InternalAsset, Pool } from '../client';
 import { checkPriceWarning } from '../pricing/checkPriceWarning';
-import { QuoteType } from '../quoting/Quoter';
 
-export default async function getPoolQuote(
-  srcAsset: InternalAsset,
-  destAsset: InternalAsset,
-  swapInputAmount: bigint,
-  fees: SwapFee[],
-  start: number,
-): Promise<{ response: QuoteQueryResponse & { quoteType: QuoteType }; duration: number }> {
-  const includedFees: SwapFee[] = [...fees];
+export default async function getPoolQuote({
+  srcAsset,
+  destAsset,
+  swapInputAmount: originalSwapInputAmount,
+  limitOrders,
+  boostFeeBps,
+  brokerCommissionBps,
+  pools,
+}: {
+  srcAsset: InternalAsset;
+  destAsset: InternalAsset;
+  swapInputAmount: bigint;
+  brokerCommissionBps?: number;
+  limitOrders?: LimitOrders;
+  boostFeeBps?: number;
+  pools: Pool[];
+}): Promise<QuoteQueryResponse> {
+  const includedFees = [];
+  let swapInputAmount = originalSwapInputAmount;
 
-  const quote = await getSwapRate({ srcAsset, destAsset, amount: swapInputAmount });
+  const brokerFee =
+    brokerCommissionBps && getPipAmountFromAmount(swapInputAmount, brokerCommissionBps);
+
+  if (brokerFee) {
+    includedFees.push(buildFee(srcAsset, 'BROKER', brokerFee));
+    swapInputAmount -= brokerFee;
+  }
+
+  if (boostFeeBps) {
+    const boostFee = getPipAmountFromAmount(swapInputAmount, boostFeeBps);
+    includedFees.push(buildFee(srcAsset, 'BOOST', boostFee));
+    swapInputAmount -= boostFee;
+  }
+
+  const { egressFee, ingressFee, networkFee, ...quote } = await getSwapRateV2({
+    srcAsset,
+    destAsset,
+    amount: swapInputAmount,
+    limitOrders,
+  });
+
+  swapInputAmount -= ingressFee.amount;
+
+  const minimumEgressAmount = await getMinimumEgressAmount(destAsset);
+
+  if (quote.outputAmount === 0n) {
+    if (networkFee.amount === 0n) {
+      // this shouldn't happen because we check before but i'll keep it here anyway
+      throw ServiceError.badRequest('swap amount is lower than ingress fee');
+    }
+
+    throw ServiceError.badRequest(
+      `egress amount (${quote.outputAmount}) is lower than minimum egress amount (${minimumEgressAmount})`,
+    );
```

Comment:
this is suppose to throw when `quote.outputAmount < minimumEgressAmount` instead or?

Reply:
The new RPC returns an output amount of 0 and an egress fee of >0